frontend/src/editor/Cell.tsx

```diff
@@ -143,6 +145,19 @@ const CellComponent = (
 
   const needsRun = edited || interrupted;
   const loading = status === "running" || status === "queued";
+  // output may or may not be refreshed while a cell is running, so
+  // we need to check if an output was received
+  const outputReceived =
+    output !== null &&
+    runStartTimestamp !== null &&
+    output.timestamp > runStartTimestamp;
+  const outputStale =
```

Comment:
@mscolnick fyi decoupled output staleness from loading status. Reading your PR I realized it was incorrect for me to modify `loading`, since it's used to control runnability of cells